### PR TITLE
Add forwarder demo

### PR DESCRIPTION
**Pull request by C-o-r-E**
This provides a demo for using nng_device to create a forwarder or proxy.

### CODE REVIEW

File: demo/pubsub_forwarder/pubsub_forwarder.c
Comment (gdamore) [inline demo/pubsub_forwarder/pubsub_forwarder.c:49]: This is a super silly nit, but I don't use these style of block comments anywhere else in this code base.

If you're willing, I'd appreciate if you could convert them to C++ style just for consistency.

I won't hold the PR up for this, because at least on the face of it, it is very nice and I'd like to see it land before the next release, but I will wait to give you a chance to respond (and possibly update) before I merge.

**Review by gdamore — APPROVED** [review]
Modulo stylistic nits, it looks great.

Still if you're willing to make the minor style fixes for consistency, I'd be even more grateful.

File: demo/pubsub_forwarder/pubsub_forwarder.c
Comment (gdamore) [inline demo/pubsub_forwarder/pubsub_forwarder.c:33]: If you use clang-format using the .clang-format I've supplied, you'll see that for if statements like this the { should be on the same line.

File: demo/pubsub_forwarder/README.adoc
Comment (gdamore) [inline demo/pubsub_forwarder/README.adoc:2]: Excellent documentation for this addition!  Thank you very much!

**Comment by gdamore** [comment]
I will try to merge this within the week, or sooner if you respond and ping me. :-) The only reason I'm not merging now is to give you a chance to make it even better by addressing the stylistic/consistency nits I pointed out. :-)

File: demo/pubsub_forwarder/pubsub_forwarder.c
Comment (C-o-r-E) [reply demo/pubsub_forwarder/pubsub_forwarder.c:49]: No problem. Will change them to match

File: demo/pubsub_forwarder/README.adoc
Comment (C-o-r-E) [reply demo/pubsub_forwarder/README.adoc:2]: Thanks! I tried to follow the other demos as an example.

**Comment by C-o-r-E** [comment]
I think that last commit takes care of the nits that needed picking. Let me know if there is anything else.

**Pull request merged**
